test/pack/universal-generator/universalGenerator.test.ts

```
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    GitHubRepoRef,
+    ProjectFile,
+} from "@atomist/automation-client";
+import { GeneratorRegistration } from "@atomist/sdm";
+import * as assert from "assert";
+import { UniversalTransform } from "../../../lib/pack/universal-generator/generatorSupport";
+import {
+    AssertGeneratorResult,
+    assertUniversalGenerator,
+} from "../../../lib/pack/universal-generator/test/assertGenerator";
+
+export const SpringGeneratorRegistration: GeneratorRegistration<{ name: string }> = {
+    name: "SpringBootGenerator",
+    intent: "create spring",
+    startingPoint: GitHubRepoRef.from({ owner: "atomist-seeds", repo: "spring-rest", branch: "master" }),
+    parameters: {
+        name: {},
+    },
+    transform: async (p, papi: any) => {
+        await p.addFile("params", `${papi.parameters.name} was here`);
+    },
+};
+
+export const Trans1UniversalTransform: UniversalTransform<{ firstName: string }> = {
+    parameters: {
+        firstName: {},
+    },
+    test: async p => true,
+    transforms: async (p, papi: any) => {
+        await p.addFile("trans1", `${papi.parameters.firstName} ${papi.parameters.name} was here`);
+    },
+};
+
+describe("universalGenerator", () => {
+
+    it("should fail to generate project with missing initial parameters", async () => {
+
+        const params = {
+            target: {
+                owner: "sdm-org",
+                repo: "test",
+            },
+        };
+
+        try {
+            await assertUniversalGenerator(
+                SpringGeneratorRegistration,
+                Trans1UniversalTransform,
+                params);
+            assert.fail();
+        } catch (e) {
+            assert(e.message === "Parameter 'name' required but missing in invocation to 'SpringBootGenerator'");
+        }
+
+    }).timeout(10000);
```

So when testing the seed is still cloned, etc.?

---

Shouldn't this one fail fast and therefore not need the additional time?